When build fails, fail and don't try to do other stuff :-)

Running a build which had a compile failure in it. It says
```
/home/jenkins/openjdk-build/build-farm/workspace/build/src//make/Main.gmk:109: recipe for target 'hotspot-only' failed
make: *** [hotspot-only] Error 2
OpenJDK make failed, archiving make failed logs
```
All good so far, except that it continues and tries to create a tarball which produces nothing which may slightly confuse the user. This should probably skip the gzip step leaving the failure messages as clear as possible at the end of the log:
```
/home/jenkins/openjdk-build/build-farm/workspace/build/src//make/Main.gmk:109: recipe for target 'hotspot-only' failed
make: *** [hotspot-only] Error 2
OpenJDK make failed, archiving make failed logs
Copying build.log to AdoptOpenJDKLogsArchive
Archiving the build OpenJDK image and compressing with gzip

real	0m0.009s
user	0m0.011s
sys	0m0.004s
Your final archive was created at /home/jenkins/openjdk-build/build-farm/workspace/build/src/build/linux-aarch64-normal-server-release/OpenJDK.tar.gz
Moving the artifact to /home/jenkins/openjdk-build/build-farm/workspace/target/
Failed to make the JDK, exiting
```

--- a/sbin/build.sh
+++ b/sbin/build.sh
@@ -1409,10 +1409,19 @@ createArchive() {
   createOpenJDKArchive "${repoLocation}" "OpenJDK"
   archive="${PWD}/OpenJDK${archiveExtension}"
 
+  archiveTarget=${BUILD_CONFIG[WORKSPACE_DIR]}/${BUILD_CONFIG[TARGET_DIR]}/${targetName}
+
   echo "Your final archive was created at ${archive}"
 
   echo "Moving the artifact to ${BUILD_CONFIG[WORKSPACE_DIR]}/${BUILD_CONFIG[TARGET_DIR]}"
-  mv "${archive}" "${BUILD_CONFIG[WORKSPACE_DIR]}/${BUILD_CONFIG[TARGET_DIR]}/${targetName}"
+  mv "${archive}" "${archiveTarget}"
+
+  if [ -f "$archiveTarget" ] && [ -s "$archiveTarget" ] ; then
+    echo "archive done."
+  else
+    echo "[ERROR] ${targetName} failed to be archived"
+    exit 1
+  fi
 }
 
 # Create a Tar ball

--- a/sbin/common/common.sh
+++ b/sbin/common/common.sh
@@ -129,7 +129,7 @@ createOpenJDKArchive()
   if which pigz > /dev/null 2>&1; then
     COMPRESS=pigz
   fi
-  echo "Archiving the build OpenJDK image and compressing with $COMPRESS"
+  echo "Archiving and compressing with $COMPRESS"
 
   EXT=$(getArchiveExtension)
 

--- a/sbin/common/common.sh
+++ b/sbin/common/common.sh
@@ -129,7 +129,7 @@ createOpenJDKArchive()
   if which pigz > /dev/null 2>&1; then
     COMPRESS=pigz
   fi
-  echo "Archiving the build OpenJDK image and compressing with $COMPRESS"
+  echo "Archiving the output and compressing with $COMPRESS"
 
   EXT=$(getArchiveExtension)
 

